### PR TITLE
add With function

### PR DESCRIPTION
**Pull request by kopiczko**
This allows to specify contextual information in the scope of logging.

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -65,6 +65,12 @@ type logger struct {
 	Logger kitlog.Logger
 }
 
+func (l *logger) With(keyvals ...interface{}) Logger {
+	return &logger{
+		Logger: kitlog.With(l.Logger, keyvals...),
+	}
+}
```
Comment (kopiczko) [inline logger.go]: This is implementation, the rest is testing :D

File: logger_test.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	{
+		wfieldValue := "test ctx field value"
+
+		out.Reset()
+		childLog.Log("msg", "whats up bro?")
+
+		var got map[string]string
+		json.Unmarshal(out.Bytes(), &got)
+
+		wcaller := "github.com/giantswarm/micrologger/logger_test.go:39"
```
Comment (xh3b4sd) [inline logger_test.go]: The test method deserves a comment because of this with a fat `NOTE`.

File: logger_test.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+	var log Logger
+	{
+		c := DefaultConfig()
+		{
+			c.IOWriter = out
+		}
+		log, err = New(c)
+		if err != nil {
+			panic(err)
```
Comment (xh3b4sd) [inline logger_test.go]: I would prefer the primitives of the testing package instead of pure panic.

File: spec.go [hunk 4]
```diff
@@ -6,4 +6,9 @@ type Logger interface {
 	// Log takes a sequence of alternating key/value pairs which are used to
 	// create the log message structure.
 	Log(v ...interface{}) error
+
+	// With returns a new contextual logger with keyvals appended to those
+	// passed to calls to Log. If logger is also a contextual logger
+	// created by With keyvals is appended to the existing context.
```
Comment (rossf7) [inline spec.go]: "it is appended"

Comment (kopiczko) [inline spec.go]: @rossf7 I think it should rather be (missing comma) `If logger is also a contextual logger created by With, keyvals is appended to the existing context.` This is the `keyvals` which is appended.

**Comment by kopiczko** [comment]
@xh3b4sd this is to solve this https://github.com/giantswarm/tprstorage/pull/1#discussion_r130617065

File: spec.go (hunk 4 above)
Comment (rossf7) [inline spec.go]: Yes with the comma it's clear.

File: logger_test.go (hunk 2 above)
Comment (kopiczko) [inline logger_test.go]: done https://github.com/giantswarm/micrologger/pull/5/commits/063538eed98e473910a4e1e3b8ad693a39e7ebd8

File: spec.go (hunk 4 above)
Comment (kopiczko) [inline spec.go]: Done https://github.com/giantswarm/micrologger/pull/5/commits/32e2b25a4c359d490d18780053ff4e5b6ff4c09e

**Pull request merged**